scripts/08-split.py

```diff
@@ -0,0 +1,56 @@
+import os
+import tarfile
+from pathlib import Path
+
+import pandas as pd
+
+
+def preprocess_file(filename: str):
+    print('reading dataset from {}'.format(filename))
+    df = pd.read_csv(filename, index_col=0)
+    df = df[~df["filename"].str.lower().str.contains("test")]
+    df = df.dropna().drop_duplicates(subset=df.columns.difference(['filename']))
+    df = df[(df.ncss > 20) & (df.ncss < 100) & (df.npath_method_avg < 100000.00)].copy()
+    return df
+
+
+if __name__ == '__main__':
```

Comment:
Quite hard to analyze long sequence of commands. I think it better be decomposed into several functions, as `process_file` above.